specutils/manipulation/replace_segment.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+
+from scipy.interpolate import CubicSpline
+
+from ..spectra import Spectrum1D
+
+
+def spline_replace(spectrum, spline_knots, extrapolation_treatment='data_fill'):
+    """
+    Generates a new spectrum with a region replaced by a smooth spline.
+
+    Parameters
+    ----------
+
+    spectrum : `~specutils.Spectrum1D`
+        The spectrum to be modified.
+
+    spline_knots : array of `~astropy.units.Quantity`
+        List of spectral axis values to be used as spline knots.
+        They all should share the same units, which can be different
+        from the units of the input spectrum spectral axis.
+
+    extrapolation_treatment : str
+        What to do with data off the edges of the region encompassed by
+        the spline knots. Default is ``'data_fill'`` to have points filled
+        with the input flux values. ``'zero_fill'`` sets them to zero.
+
+    Returns
+    -------
+    spectrum : `~specutils.Spectrum1D`
+        The spectrum with the region replaced by spline values, and data
+        values or zeros outside the spline region. The spectral axis will
+        have the same units as the spline knots.
+
+    Notes
+    -----
+    The uncertainty attribute, if present in the input spectrum, is not
+    carried over into the output.
```

Comment:
I think it may make sense to just interpolate the uncertainty *exactly* the wsame way as the flux, right? It might be this should be an option (``interpolate_uncertainty=True``) since sometimes that's not the right thing, but as a "sometimes reasonable default" I think it might be a good idea

Reply:
I added uncertainty handling to the code.